### PR TITLE
Liu2021swin_2 initial draft

### PR DESCRIPTION
**Pull request by ywen666**
TODO: Shifted Window based Self-Attention, experiments

### CODE REVIEW

File: _summaries/liu2021swin_2.md [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+## Abstract
+* Transformer is designed for text domain. The high resolution of pixels in images prevent smooth application of transformer in visual domain. 
+* Semantic segmentation that require dense prediction at the pixel level, and this would be intractable for Transformer ($$O(n^2)$$).
+* This paper proposed to a hierarchical Transformer whose representation is computed with shifted windows (SWIN).
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:13]: t̶o̶

File: _summaries/liu2021swin_2.md [hunk 2]
```diff
@@ -1,19 +1,45 @@
...
-* Bullets
-* To highlight the core concepts
+## Motivation
+To reduce complexity, Swin Transformer constructs a hierarchical representation by starting from small-sized patches (outlined in gray) and gradually merging neighboring patches in deeper
+Transformer layers. 
+
+<img width="650px" src="liu2021swin_2a.png"/>  
+
+The linear computational complexity is achieved by computing self-attention locally within non-overlapping windows that partition an image (outlined in red). The number of patches in each
+window is fixed, and thus the complexity becomes linear to image size.
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:25]: "linear in image size"

File: _summaries/liu2021swin_2.md [hunk 3]
```diff
@@ -1,19 +1,45 @@
...
+
+A key design element of Swin Transformer is its shift of the window partition between consecutive self-attention layers. The shifted windows bridge the windows of the preceding layer, providing connections among them that significantly enhance modeling power. 
+
+<img width="650px" src="liu2021swin_2b.png"/>  
+
+## Overall Architecture
+
+Swin Transformer first splits an input RGB image into non-overlapping patches by a patch splitting module. Each patch is treated as a “token” and its feature is set as a concatenation of the raw pixel RGB values. A linear embedding layer is applied on this raw-valued feature to project it to an arbitrary dimension (denoted as C).
+
+Swin Transformer blocks are applied on these patch tokens. The Transformer blocks maintain the number of tokens ($$\frac{H}{4} \times \frac{W}{4}$$), and together with the linear embedding are referred to as “Stage 1”.
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:35]: Multiple Swim Transformer blocks ....

File: _summaries/liu2021swin_2.md [hunk 4]
```diff
@@ -1,19 +1,45 @@
...
+
+The first patch merging layer concatenates the features of each group of 2 × 2 neighboring patches, and applies a linear layer on the 4C-dimensional concatenated features. This reduces the number of tokens by a multiple of 2×2 = 4 (2× downsampling of resolution), and the output dimension is set to 2C. Swin Transformer blocks are applied afterwards for feature transformation, with the resolution kept at $$\frac{H}{8} \times \frac{W}{8}$$. This first block of patch merging
+and feature transformation is denoted as “Stage 2”.
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:38]: Stage 2 is followed by stages 3 and 4 that have the same architecture except for the number of Swin Transformer blocks and the resolution of the windows.

File: _summaries/liu2021swin_2.md [hunk 5]
```diff
@@ -57,4 +57,13 @@ To reduce the latency, the authors  propose a more efficient batch computation a
 
 <img width="500px" src="liu2021swin_2d.png"/>  
 
-## Experiements
\ No newline at end of file
+## Experiements
+
+The paper conducted a thorough empirical evaluation on ImageNet-1k classification, object detection and semantic segmentation. The results showed that Swin-Transformer outperforms ResNet and Vit when used as the backbone model.
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:62]: ViT

File: _summaries/liu2021swin_2.md [hunk 6]
```diff
@@ -57,4 +57,13 @@ To reduce the latency, the authors  propose a more efficient batch computation a
...
+
+<img width="500px" src="liu2021swin_2e.png"/>
+<img width="500px" src="liu2021swin_2f.png"/>
+<img width="500px" src="liu2021swin_2g.png"/>
+
+## TL;DR
+Swin-Transformer — A hierarchical Transformer whose representation is computed with Shifted windows. This hierarchical architecture has the flexibility to model at various scales and has linear computational complexity with respect to image size. It achieved SOTA on various of vision benchmarks when used as the backbone model.
```
Comment (ayushchauhan) [inline _summaries/liu2021swin_2.md:69]: "various o̶f̶ vision benchmarks"

**Comment by ayushchauhan** [comment]
Very good summary overall. I don't have much to add to it, all the important points have been covered and explained clearly.

File: _summaries/liu2021swin_2.md [hunk 7]
```diff
@@ -1,19 +1,45 @@
...
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+## Abstract
+* Transformer is designed for text domain. The high resolution of pixels in images prevent smooth application of transformer in visual domain. 
+* Semantic segmentation that require dense prediction at the pixel level, and this would be intractable for Transformer ($$O(n^2)$$).
```
[the remark this answers is not in the document]
Comment (ywen666) [reply _summaries/liu2021swin_2.md:12]: Done

File: _summaries/liu2021swin_2.md (hunk 1 above)
Comment (ywen666) [reply _summaries/liu2021swin_2.md:13]: done

**Review by zhaoyue-zephyrus — APPROVED** [review]

**Pull request merged**